src/ioctl.c

```diff
@@ -371,7 +372,12 @@ ioctl_decode(struct tcb *tcp)
 	case 'I':
 		return inotify_ioctl(tcp, code, arg);
 	case 'K':
-		return kd_ioctl(tcp, code, arg);
+		/* FIXME: this is ugly, and wont work forever */
+		if (_IOC_NR(code) >= _IOC_NR(KIOCSOUND)) {
```

Comment:
Though I didn't finish revising after reviewing, the patch I proposed in https://lists.strace.io/pipermail/strace-devel/2022-March/010996.html adds the ability to distinguish the type of file descriptors to strace. Just comment.